source/PlayerInfo.cpp

```diff
@@ -1398,16 +1401,25 @@ void PlayerInfo::SetShipOrder(const vector<shared_ptr<Ship>> &newOrder)
 
 // Find out how attractive the player's fleet is to pirates. Aside from a
 // heavy freighter, no single ship should attract extra pirate attention.
-pair<double, double> PlayerInfo::RaidFleetFactors() const
+pair<double, double> PlayerInfo::RaidFleetFactors(const System *system) const
 {
 	double attraction = 0.;
 	double deterrence = 0.;
+	bool defaultAttraction = !system || !system->GetRaiders() || !system->GetRaiders()->ScoutsCargo();
+	double emptyCargoAttraction = defaultAttraction ? 1. : system->GetRaiders()->EmptyCargoAttraction();
 	for(const shared_ptr<Ship> &ship : Ships())
 	{
 		if(ship->IsParked() || ship->IsDestroyed())
 			continue;
 
-		attraction += ship->Attraction();
+		if(defaultAttraction || !ship->Cargo().Size())
+			attraction += ship->Attraction();
+		else
+		{
+			double emptyCargoRatio = 1. - (ship->Cargo().Free() / ship->Cargo().Size());
```

Comment:
I'm pretty sure we don't want int division here
```suggestion
			double emptyCargoRatio = 1. - ship->Cargo().FreePrecise() / ship->Cargo().Size();
```